### ISSUE TITLE
El Capitan cannot locate the binaries sift and unable to load libsiftgpu.so

### ISSUE THREAD

**Issue by uwe22** [luckybulldozer/VisualSFM_OS_X_Installer#25]
getting the Userinterface to appear on El Capitan,
I cannot get the software to work.
With or without GPU, with different sifts all installed in the same directory as VisualSFM

when I want to detect features

I always get the message: cannot locate sift binaries

SIFT: 0000, 1022x1774, 	ERROR1
ERROR: unable to locate sift binary.
FAQ at http://ccwu.me/vsfm/doc.html#errors

###########-------timing------#########
1 Feature Detection finished, 0 sec used
#############################

3 pairs to compute match
NOTE: using 2 matching workers

###########-------timing------#########
3 Image Match finished, 0 sec used
#############################

----------------------------------------------------------------
Compute Missing Pairwise Matching, finished
Totally 0.000 seconds used

or when I disable GPU:

0: 120p_V_042
image_size:	1022x1774
1: 120p_V_041
image_size:	1118x1765
2: 120p_V_040
image_size:	1035x1776
Loading image pixel data ...done in 0s
ERROR: unable to load  libsiftgpu.so
ERROR: unable to get function address
SIFT #:	ERROR1

the nv.ini file seems not to be read at all, what ever I put in there does not 
affect the bevaviour of the program on start. parameter changes not seem to be saved there.

Please help.

**Comment by jstenner** [comment]
Same issue here on El Capitan. Built fine, libsiftgpu.so is in the bin directory, interface appears, able to load an image set, but fails to find libsiftgpu.so when I try to start a match.

**Comment by whaus7** [comment]
Arg, same thing here with El Capitan. Either of you guys ever figure it out?

**Comment by wiltongorske** [comment]
No. I couldn't even get it working on Raspbian! I may partition part of my drive for Ubuntu or even find a Windows laptop.

📱

**Comment by whaus7** [comment]
Dan the Man Monaghan just got back to me about this.

From Dan:
"Double check, I had the same problem today when I tried a fresh install and it was that there was no LD_LIBRARY_PATH set up, once I'd done that it was fine."

Trying to figure out how to do this right now. I'm just a front-end web monkey. My systems admin skills are poor at best :)

**Comment by jstenner** [comment]
I have it working on a nMac Pro that has been "upgraded" to El Capitan but still has gtk+ 2.24.25. I mistakenly updated brew on my Macbook Pro/El Crapitan system which updated to gtk+ 2.24.28 (which no longer has X11 support) and it no longer works. A friend has a similar problem with newer machines that have fresh Homebrew installs that put the latest version of gtk+. We've tried to roll these machines back to the older version of gtk+ but despite trying all sorts of shenanigans, it just won't work. They've changed the way to reinstall versions and the current repository doesn't have any older gtk+ formulas. https://github.com/Homebrew/homebrew-versions

Since I have one machine that works, I gave up....

**Comment by whaus7** [comment]
Ugh, gross... well good news is I seem to be making progress. Think my images are just poor/not enough because I'm only getting like 2-3 camera projections (or however this madness is working) and then its saying 'Failed to find two images for initialization'

Any experience with this?

**Comment by luckybulldozer** [comment]
That definitely sounds like a quality problem.  I'll often shoot hundreds of images on DLSR, extremely sharp, well exposed images too, low noise etc...  But to get things working work with a well shot smaller set first, as matching 200-300 images on a new MacPro is still pretty laborious.  

As for the brew version... you can alway do a
```
brew remove gtk+ 
brew install https://raw.githubusercontent.com/Homebrew/homebrew/99126a50c96b3c832d72f4531c116271f543eded/Library/Formula/gtk+.rb
```

Annoying, but it should work.

**Comment by whaus7** [comment]
Wow.. 200-300? Is it possible to generate a good modal for a very basic single object with 50-60 images? Guess I'll find out soon lol. Now computing dense reconstruction is throwing an error

Running Yasutaka Furukawa's CMVS tool...
cmvs /Users/whausman/Desktop/test-object.cmvs/00/ 50 8
ERROR: the above CMVS command failed!

Any idea? Thanks again for all the help Dan!

**Comment by whaus7** [comment]
Well I definitely see cmvs, genOption, or pmvs2 all in the same folder as VisualSFM exe per the FAQ's page.

**Comment by whaus7** [comment]
So.. dense construction seems to be working if I select NVM -> CMP-MVS instead of CMP -> CMVS for the output file type

Is this ok to do? Now after dense construction finishes if I hit tab for viewing dense 3d points. all points  disappear and it just goes blank white

Anyone have this kind of behavior before?

**Comment by whaus7** [comment]
Rock on... did a fresh install and I'm in business!

**Comment by thebussurfer** [comment]
Hello, I am having the same problem with this

ERROR: unable to load  libsiftgpu.so
ERROR: unable to get function address
SIFT: 0000, 1944x2592, 	ERROR1
ERROR: unable to locate sift binary.
FAQ at http://ccwu.me/vsfm/doc.html#errors

I saw that whaus7 fixed it by setting up LB_LIBRARY_PATH but I don't know how to do that :(

Any help would be amazing!
thanks

**Comment by indy500driver** [comment]
Would I use nano to set up the LB_LIBRARY_PATH in my bash profile?

**Comment by Saandji** [comment]
To setup LB_LIBRARY_PATH, simply:
1. open terminal
2. use your favorite tool to edit ~/.bash_profile (for example: nano ~/.bash_profile)
3. paste two string at the bottom: (replace YOUR_PATH_TO_SFM with your path, where you have SFM installed) something like: /Users/somebody/Downloads/
export PATH=YOUR_PATH_TO_SFM/VisualSFM_OS_X_Installer-master/vsfm/bin:$PATH
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/YOUR_PATH_TO_SFM/VisualSFM_OS_X_Installer-master

hope this helps

**Comment by blaaaaaaaam** [comment]
Hi,

I'm encountering this issue too. I've added the LB_LIBRARY_PATH to my bash profile using what it said to at the end of the installer.

I'm trying to run it off cpu. I'm using a Retina Macbook that has the integrated graphics IRIS pro card (no dedicated gpu).

I get the standard error code:
ERROR: unable to load libsiftgpu.so
ERROR: unable to get function address
SIFT: 0000, 1944x2592, ERROR1
ERROR: unable to locate sift binary.
FAQ at http://ccwu.me/vsfm/doc.html#errors

**Comment by Duuk15** [comment]
Hi, I too am trying to use VisualFSM on Mac OS X ~ El Capitan

I'm trying to follow the work flow of Jesse at http://wedidstuff.heavyimage.com/index.php/2013/07/12/open-source-photogrammetry-workflow/ so that I can eventually 3D print a paperweight model of a building at my alma mater. I'm currently working with a smaller set of images of a statue instead until I get a handle on what I'm doing.  Any help is greatly appreciated.  I'm not a computer programmer and I'm impressed I've made it to  Step B before encountering problems to trouble shoot.

I've tried to run @luckybulldozer gtk+ uninstall and install, but the install appears to be directing gtk+ 2.24.28 to be installed and not gtk+ 2.24.25, hence I'm still getting the same errors as @thebussurfer that I'm trying to correct. 

Thank you for your help.

**Comment by gpanders** [comment]
I was able to finally get this to work by doing the following:

1) Modifying the PATH and LD_LIBRARY_PATH like the end of the install scripts says and @Saandji pointed out
2) In the top level installer folder, copy or move the `SiftGPU` folder to `vsfm/bin/sift`
3) Copy or move the `vsfm/bin/libsiftgpu.so` file to the top level installer folder

**Comment by jluna79** [comment]
Thank you @gpanders, worked like a charm! +1

**Comment by H0icky** [comment]
@gpanders it works for me

1. Modifying the PATH and LD_LIBRARY_PATH like the end of the install scripts says and @Saandji pointed out
2. Copy or move the `vsfm/bin/libsiftgpu.so` file to the top level installer folder

**Comment by elipeters** [comment]
@gpanders no shut down all the ... detention level!

1. follow this [https://github.com/luckybulldozer/VisualSFM_OS_X_Installer]
2. modify or create bash_profile @Saandji 
3. then move all the created files from the `vsfm/bin/` to the top level installer folder `VisualSFM_OS_X_Installer-master`

I also moved `SiftGPU` to the bin folder and renamed this to `sift` though not sure if that was necessary.

**Comment by raisonm** [comment]
Was there any other process any one went through? this still isn't working for me. I've tried reloading the stuff 3 times and moving all the files in to place as described. I can't seem to get it to work at all
-__-

I checked the dependencies for libsiftgpu too and I believe everything is in place.
